src/components/v2/values/common/chartValues.api.ts

```diff
@@ -1,79 +1,201 @@
-            } else {
-                setSelectedVersionUpdatePage(result[0]);
-            }
-        } else if (valueUpdateRequired) {
-            setSelectedVersionUpdatePage(result[0]);
-        }
+        dispatch({
+            type: 'multipleOptions',
+            payload: {
+                isLoading: false,
+                chartVersionsData: result,
+                selectedVersion: _currentVersion.id,
+                selectedVersionUpdatePage: _currentVersion,
+            },
+        })
     } catch (err) {
-        showError(err);
-    } finally {
-        setLoading && setLoading(false);
+        showError(err)
+        dispatch({
+            type: 'isLoading',
+            payload: false,
+        })
     }
 }
 
 export async function getChartValuesList(
     id: number,
     setChartValuesList: React.Dispatch<React.SetStateAction<any>>,
-    setChartValues: React.Dispatch<React.SetStateAction<any>>,
-    setLoading?: React.Dispatch<React.SetStateAction<boolean>>,
+    handleChartValuesSelection?: (chartValues) => void,
     initId?: number,
     installedAppVersionId = null,
 ) {
-    setLoading && setLoading(true);
     try {
-        const { result } = await getChartValuesCategorizedListParsed(id, installedAppVersionId);
-        setChartValuesList(result);
-        if (installedAppVersionId) {
-            setChartValues({
+        const { result } = await getChartValuesCategorizedListParsed(id, installedAppVersionId)
+        setChartValuesList(result)
+        if (installedAppVersionId && handleChartValuesSelection) {
+            handleChartValuesSelection({
                 id: initId,
                 kind: 'EXISTING',
-            });
+            })
         }
     } catch (err) {
-        showError(err);
-    } finally {
-        setLoading && setLoading(false);
+        showError(err)
     }
 }
 
 export async function getChartRelatedReadMe(
     id: number,
-    setFetchingReadMe: React.Dispatch<React.SetStateAction<boolean>>,
-    setActiveReadMe: React.Dispatch<React.SetStateAction<string>>,
+    currentFetchedReadMe: Map<number, string>,
+    dispatch: (action: ChartValuesViewAction) => void,
 ) {
     try {
-        setFetchingReadMe(true);
-        const { result } = await getReadme(id);
-        setActiveReadMe(result.readme);
-        setFetchingReadMe(false);
+        dispatch({ type: 'fetchingReadMe', payload: true })
+        const { result } = await getReadme(id)
+        const _fetchedReadMe = currentFetchedReadMe
+        _fetchedReadMe.set(id, result.readme)
+
+        dispatch({
+            type: 'multipleOptions',
+            payload: {
+                fetchingReadMe: false,
+                fetchedReadMe: _fetchedReadMe,
+                isReadMeAvailable: !!result.readme?.trim(),
+            },
+        })
     } catch (err) {
-        showError(err);
-    } finally {
-        setFetchingReadMe(false);
+        showError(err)
+        dispatch({
+            type: 'multipleOptions',
```

Comment:
Can we make all these types constants?

Reply:
Done. Created an enum - `ChartValuesViewActionTypes` for the same.